### PR TITLE
Ensure Docker composition starts up successfully

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR makes the following changes so that the pca-gophish Docker composition is able to start up successfully at boot time:
* Installs `docker-compose` via pip in a virtual Python environment
* Updates the `pca-gophish-composition` systemd services so that it manages the pca-gophish Docker composition using the instance of `docker-compose` in the virtual Python environment
* Converts some Ansible collections to use FQCNs (most were already done, but a couple had been missed)

## 💭 Motivation and context ##

I observed that in our latest AMI created with this repo, the pca-gophish Docker composition was not starting up at instance boot and could not be manually started, either via systemd or docker-compose.  

The reason for this is the latest available version of docker-compose on Debian Buster is 1.21.0 (see https://packages.debian.org/source/buster/docker-compose), but the pca-gophish Docker composition requires version 3.7 of the docker-compose file format, which is supported by docker-compose 1.22.0 and later.  To work around this issue, we create a virtualenv and install docker-compose via pip, which gives us a newer version that supports the docker-compose file format 3.7.

Related to this, I created issue #34 to undo this virtualenv workaround when it is no longer needed.

This PR is labelled "blocked" because we want to wait until https://github.com/cisagov/pca-gophish-composition/pull/28 has been merged so we can build another Staging AMI (and eventually Production) that includes the newest pca-gophish Docker composition.

## 🧪 Testing ##

I built an AMI and deployed it to a Staging environment, where I confirmed that the systemd service and corresponding pca-gophish composition were automatically started after the instance booted.  I also verified that I was able to login to Gophish in a browser on that instance.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] _All_ future TODOs are captured in issues, which are referenced
      in code comments.
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All new and existing tests pass.
* [x] https://github.com/cisagov/pca-gophish-composition/pull/28 has been merged.
* [x] A Staging AMI has been built and tested after https://github.com/cisagov/pca-gophish-composition/pull/28 has been merged.

## ✅ Pre-merge checklist ##
* [x] Finalize version to `0.1.3` by running `./bump_version.sh finalize` - see https://github.com/cisagov/pca-gophish-composition-packer/pull/33/commits/d673764ba8c203e6328ecd83ab13c971b2e51da4
* [x] Create the `0.1.3` release in GitHub - see [release `v0.1.3`](https://github.com/cisagov/pca-gophish-composition-packer/releases/tag/v0.1.3)

### CODE REVIEW

File: src/version.txt [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = "0.1.2"
+__version__ = "0.1.3+build.1"
```
Comment (jsf9k) [inline src/version.txt:1]: Why did you need to bump the build here as well?  Did you build a 0.1.3 AMI that had an error?

**Review by jsf9k — APPROVED** [review]
This looks great to me, but I have one minor question.

Another solution would be to build this on Debian Bullseye.  Bullseye is due to be released in a month, so it is pretty stable now.

File: src/version.txt (hunk 1 above)
Comment (dav3r) [reply src/version.txt:1]: I didn't- I thought that the first build should be `0.1.3+build.1` and when everything is working, it would end up at `0.1.3`, but based on your question, I'm thinking that's not correct.

**Comment by dav3r** [comment]
> Another solution would be to build this on Debian Bullseye. Bullseye is due to be released in a month, so it is pretty stable now.

I think this fix will hold us over for a little while.  Don't you think we should start by moving [`cisagov/skeleton-packer`](https://github.com/cisagov/skeleton-packer) over to Bullseye and then go from there?

File: src/version.txt (hunk 1 above)
Comment (jsf9k) [reply src/version.txt:1]: I do `0.1.3-rc.1` and `0.1.3-rc.2`, for example, for the staging AMI builds and then finalize that to `0.1.3` when everything is working and it's time to build the production AMI.

The build numbers are for when you don't change the code but rebuild the AMI, for example when freshening the AMIs for the monthly redeploy.

**Comment by jsf9k** [comment]
> > Another solution would be to build this on Debian Bullseye. Bullseye is due to be released in a month, so it is pretty stable now.
> 
> I think this fix will hold us over for a little while. Don't you think we should start by moving [`cisagov/skeleton-packer`](https://github.com/cisagov/skeleton-packer) over to Bullseye and then go from there?

I'd be OK with allowing specific packer repos that need it to jump to Bullseye before it is released.  I'd probably hold back [`cisagov/skeleton-packer`](https://github.com/cisagov/skeleton-packer) until Bullseye officially drops.

I'm also OK with this solution; I just mentioned jumping to Bullseye as an alternative.

**Review by mcdonnnj — APPROVED** [review]
Reasonable workaround. Thanks for doing the research on what was going wrong 🏆

File: src/version.txt (hunk 1 above)
Comment (dav3r) [reply src/version.txt:1]: Got it- I will try to remember to follow that same scheme in the future.

Comment (jsf9k) [reply src/version.txt:1]: > Got it- I will try to remember to follow that same scheme in the future.

I'll be watching. :eyes:

Comment (dav3r) [reply src/version.txt]: Proof: https://github.com/cisagov/pca-gophish-composition-packer/pull/33/commits/0196468386b8679034579782b5c1f9d5380d95c8

Comment (dav3r) [reply src/version.txt]: [Behold!](https://github.com/cisagov/pca-gophish-composition-packer/releases/tag/v0.1.3%2Brc.1)

**Comment by dav3r** [comment]
@JCantu248 - I have deployed a freshly built AMI from [release `0.1.3-rc.1`](https://github.com/cisagov/pca-gophish-composition-packer/releases/tag/v0.1.3%2Brc.1) to your Staging environment.  Please verify that everything looks correct out there from a Gophish perspective and then reply here so that the finalized release can be created and this PR can be merged.  Thanks!

**Review by JCantu248 — APPROVED** [review]
It looks good to me. I approve merging these changes
